pkg/storaged/stratis-details.jsx

```diff
@@ -207,6 +216,153 @@ export const StratisPoolDetails = ({ client, pool }) => {
+                    }
+                }
+            });
+        });
+    }
+
+    function remove_passphrase() {
+        dialog_open({
+            Title: _("Remove passphrase?"),
+            Body: <div>
+                <p className="slot-warning">{ fmt_to_fragments(_("Passphrase removal may prevent unlocking $0."), <b>{pool.Name}</b>) }</p>
+            </div>,
+            Action: {
+                DangerButton: true,
+                Title: _("Remove"),
+                action: function (vals) {
+                    return pool.UnbindKeyring().then(std_reply);
+                }
+            }
+        });
+    }
+
+    function add_tang() {
+        return with_keydesc(client, pool, (keydesc, keydesc_set) => {
+            dialog_open({
+                Title: _("Add Tang keyserver"),
+                Fields: [
+                    TextInput("tang_url", _("Keyserver address"),
+                              {
+                                  validate: validate_url
+                              }),
+                    Skip("medskip",
+                         {
+                             visible: () => !keydesc_set
+                         }),
+                    PassInput("passphrase", _("Disk passphrase"),
+                              {
+                                  visible: () => !keydesc_set,
+                                  validate: val => !val.length && _("Passphrase cannot be empty"),
+                                  explanation: _("Adding a keyserver requires unlocking the pool. Please provide the existing pool passphrase.")
+                              })
+                ],
+                Action: {
+                    Title: _("Add"),
+                    action: function (vals, progress) {
+                        return get_tang_adv(vals.tang_url)
+                                .then(adv => {
+                                    function bind() {
+                                        if (pool.BindClevis) {
+                                            return pool.BindClevis("tang", JSON.stringify({ url: vals.tang_url, adv })).then(std_reply);
+                                        } else {
+                                            return pool.Bind("tang", JSON.stringify({ url: vals.tang_url, adv })).then(std_reply);
+                                        }
+                                    }
+
+                                    confirm_tang_trust(vals.tang_url, adv,
+                                                       () => {
+                                                           if (vals.passphrase)
+                                                               return with_stored_passphrase(client, keydesc,
+                                                                                             vals.passphrase, bind);
+                                                           else
+                                                               return bind();
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18818-20230601-113309-0e7aeb29-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)